app/src/main/java/com/example/travellens/MessageAdapter.java

```diff
@@ -0,0 +1,114 @@
+package com.example.travellens;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.parse.ParseUser;
+
+
+import java.util.List;
+
+public class MessageAdapter extends RecyclerView.Adapter<MessageAdapter.MessageViewHolder> {
+    public static final int MESSAGE_LEFT = 0; // incoming
+    public static final int MESSAGE_RIGHT = 1; // outcoming
+    private Context context;
+    private List<Message> messages;
+
+    public MessageAdapter(Context context, List<Message> messages) {
+        this.context = context;
+        this.messages = messages;
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        if (isMe(position)) {
+            return MESSAGE_RIGHT;
+        } else {
+            return MESSAGE_LEFT;
+        }
+    }
+
+    private boolean isMe(int position) {
+        Message message = messages.get(position);
+        return message.getSender() != null && message.getSender()
+                .equals(ParseUser.getCurrentUser().getString(Post.KEY_FIREBASE_USER_ID));
+    }
+
+    @NonNull
+    @Override
+    public MessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        Context context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+
+        if (viewType == MESSAGE_LEFT) {
+            View contactView = inflater.inflate(R.layout.item_left_chat, parent, false);
```

Comment:
Similarly, I would rename the layout files with more obvious/meaningful names too.